### PR TITLE
move bundle size check to the server

### PR DESCRIPTION
**Pull request by shamit6**

### CODE REVIEW

File: packages/next-in-it-stats/src/sizeCheckBundles.ts [hunk 1]
```diff
@@ -32,39 +46,40 @@ export async function sizeCheckBundles({ outDir }: { outDir: string }): Promise<
...
+  setFormData(formData, 'packageName', packageName);
+  setFormData(formData, 'trackedFiles', JSON.stringify(trackedFiles));
+  setFormData(formData, 'fileSizes', JSON.stringify(fileSizes));
 
-  // TODO send result to github status check
+  // @ts-ignore-next-line
+  const response = await fetch(`${serverUrl.toString()}/bundle-size-check`, {
+    method: 'POST',
+    body: formData,
+  });
```
Comment (deanshub) [inline packages/next-in-it-stats/src/sizeCheckBundles.ts]: you don't need form data if you don't send files

File: packages/shared-types/documents.ts [hunk 2]
```diff
@@ -62,3 +62,17 @@ export interface FileTrackingDocument {
...
+  appId: string; // (FK)
+  commitHash: string;
+  trackedFiles: {
+    [filePath: string]: {
+      size: number;
+      status: 'passed' | 'failed';
+      reason?: string; //'size limit exceeded' | 'percentage of change exceeded';
+    };
+  };
+  status: 'passed' | 'failed';
```
Comment (deanshub) [inline packages/shared-types/documents.ts:77]: do you need this or can this be derived?

File: packages/next-in-it-stats/src/sizeCheckBundles.ts [hunk 3]
```diff
@@ -32,39 +46,40 @@ export async function sizeCheckBundles({ outDir }: { outDir: string }): Promise<
...
 
-  // if there are any relative limits, compare to the previous\master build
-  // gather all errors and print them
-  // if there are any errors, exit with error code
-  if (errors.length) {
-    console.error(errors.join('\n'));
+  if (response.status !== 200) {
```
Comment (deanshub) [inline packages/next-in-it-stats/src/sizeCheckBundles.ts]: I think we should return `406` when it's a validation issue to handle it properly and do something else for the rest

File: packages/web/package.json [hunk 4]
```diff
@@ -69,6 +69,7 @@
     "kuuid": "^1.0.1",
     "lucide-react": "^0.220.0",
     "mongoose": "^7.2.0",
+    "multimatch": "^6.0.0",
```
Comment (deanshub) [inline packages/web/package.json:74]: do we still need `multimatch` in `next-in-it-stats`?

File: packages/web/src/db/models/BundleSizeValidation.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    commitHash: {
+      type: String,
+      required: true,
+    },
+    status: {
+      type: String,
+      required: true,
+    },
+    trackedFiles: {
+      type: Schema.Types.Mixed,
```
Comment (deanshub) [inline packages/web/src/db/models/BundleSizeValidation.ts:26]: can't we have a custom schema?

File: packages/web/src/db/queries/getBundleSizeValidation.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  buildId?: string;
+  commitHash?: string;
+  status?: string;
+}
+
+export function getBundleSizeValidation(
+  { appId, status }: BundleSizeValidationFilter,
+  { limit, offset }: { limit: number; offset: number },
+) {
+  console.log('getBundleSizeValidation', {
```
Comment (deanshub) [inline packages/web/src/db/queries/getBundleSizeValidation.ts]: ?

File: packages/web/src/utils/github.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  status,
+  conclusion,
+}: // target_url,
+ReportBundleSizeStatusCheckParams) {
+  const privateKey = getOrThrow('NISSIX_GITHUB_PRIVATE_KEY').replace(/\\n/g, '\n');
+  const appId = getOrThrow('NISSIX_GITHUB_APP_ID');
+  const githubAppUrl = getOrThrow('NISSIX_GITHUB_APP_URL');
+
+  const owner = repository.split('/')[0];
+  const repoName = repository.split('/')[1];
```
Comment (deanshub) [inline packages/web/src/utils/github.ts]:
```suggestion
  const [owner, repoName] = repository.split('/');
```

File: packages/web/src/utils/github.ts [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        privateKey,
+        installationId,
+      },
+    });
+
+    const response = await installationOctokit.rest.checks.create({
+      installationId,
+      owner,
+      repo: repoName,
+      name: `bundle size check - ${packageName}`,
```
Comment (deanshub) [inline packages/web/src/utils/github.ts]:
```suggestion
      name: `${packageName} - bundle size check`,
```

File: packages/web/src/utils/github.ts [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+      head_sha,
+      status: status ?? 'completed', //queued, in_progress, or completed
+      conclusion: conclusion ?? 'success', //success, failure, neutral, cancelled, timed_out, or action_required
+      // target_url,
+      description,
+    });
+    return `Bundle size analysis status check was created, you can follow its run here: ${response.data.html_url}`;
```
Comment (deanshub) [inline packages/web/src/utils/github.ts]:
```suggestion
    return `Bundle size analysis status check was created, you can follow it's run here: ${response.data.html_url}`;
```

**Review by deanshub — APPROVED** [review]

File: packages/web/src/db/models/BundleSizeValidation.ts (hunk 5 above)
Comment (shamit6) [reply packages/web/src/db/models/BundleSizeValidation.ts:26]: There are a few annoying issue with Map in mongoose so it much more simple for now make that way

File: packages/shared-types/documents.ts (hunk 2 above)
Comment (shamit6) [reply packages/shared-types/documents.ts:77]: yes it can. but it simplify the mongo query of getting the last passed status

**Pull request merged**